src/clj/athens/self_hosted/web/presence.clj

```diff
@@ -58,6 +59,21 @@
         (common-events/build-event-accepted id max-tx)))))
 
 
+(defn username-handler
+  [datahike channel {:event/keys [id args]}]
+  (let [{:keys
+         [current-username
+          new-username]}         args
+        max-tx                   (:max-tx @datahike)
+        broadcast-username-event (common-events/build-presence-broadcast-username-event max-tx
+                                                                                        current-username
+                                                                                        new-username)]
+
+    (clients/add-client! channel new-username)
```

Comment:
should we be adding a client?
This client already exists and it's renaming.

Reply:
It works but the function should be renamed.